git-ghpr

notCalle opened the pull request:
GitHub Pull Request helper script.

First stab. Read the script before using. No input validation. Might shave your cat, or set the sun on fire.

notCalle committed:
GitHub Pull Request helper script

First stab. Read the script before using. No input validation.
Might shave your cat, or set the sun on fire.

notCalle committed:
README file for git-ghpr

notCalle committed:
Mention GPG signing in usage.

notCalle committed:
Correct mode bits

notCalle committed:
Fix invalid branch name syntax

creideiki commented:
I'm nervous about pushing things I haven't looked at. The state of the repository changes significantly between `merge --no-ff --no-commit` and `commit`, but this workflow does not allow me to see that.

I especially want to see the entire commit chain in the pull request, not just a flattened result diff.

notCalle committed:
Split commit+push phase into separate commit & push phases.

creideiki commented on git-ghpr/bin/git-ghpr:
```diff
@@ -0,0 +1,82 @@
+    git pull --ff-only || return
+    git pull --no-ff --verify-signatures --no-commit "$source_url" "$branch" || return
+    merge_base=`cat .git/ORIG_HEAD`
+    merge_head=`cat .git/MERGE_HEAD`
```
Relative paths make me nervous.

notCalle committed:
Be explicit about the git toplevel directory pathname

creideiki commented on git-ghpr/bin/git-ghpr:
```diff
@@ -0,0 +1,83 @@
+
+topdir=$(git rev-parse --show-toplevel)
+origin_url=$(git remote get-url origin)
+project=$(basename $origin_url .git)
```
"upstream", not "origin".

notCalle committed:
Parse the project url from upstream, not origin.

origin is expected to point at a fork, and upstream is expected to
point at the receiver of pull requests. In case of a maintainer only
checkout, the only remote will be the upstream remote, and origin
might not exist.

The actual project name should be parsed from the upstream URL. This
will still break if the fork has renamed their version, but solving
that requires actual GitHub API calls.

Minor README text change.
